### PR TITLE
feat: Adds new `gen_ai_features_enabled` attribute support  in `mongodbatlas_organization` resource & data sources

### PR DESCRIPTION
**Pull request by maastha**
## Description

*****NOTE: This feature is yet to be released to prod. This PR will not be merged until the feature is available in production and Atlas SDK version has been updated.  See JIRA ticket for timeline details.**

Adds support for GenAI Control Switch in Organizations settings.

Link to any related issue(s): [CLOUDP-264109](https://jira.mongodb.org/browse/CLOUDP-264109)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [X] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/organization/resource_organization_test.go [hunk 1]
```diff
@@ -161,6 +165,95 @@ func TestAccOrganizationCreate_Errors(t *testing.T) {
 	})
 }
 
+func TestAccConfigDSOrganization_basic(t *testing.T) {
```
Comment (maastha) [inline internal/service/organization/resource_organization_test.go:168]: moved data source tests in the same file to be in alignment with our latest practices

File: internal/service/project/resource_project.go [hunk 2]
```diff
@@ -785,7 +786,7 @@ func UpdateProjectTeams(ctx context.Context, teamsAPI admin.TeamsApi, projectSta
 		_, err := teamsAPI.RemoveProjectTeam(ctx, projectID, teamID).Execute()
 		if err != nil {
 			apiError, ok := admin.AsError(err)
-			if ok && *apiError.ErrorCode != "USER_UNAUTHORIZED" {
+			if ok && apiError.ErrorCode != "USER_UNAUTHORIZED" {
```
Comment (maastha) [inline internal/service/project/resource_project.go]: change from the SDK

File: internal/service/cloudbackupsnapshotexportbucket/data_source_cloud_backup_snapshot_export_buckets.go [hunk 3]
```diff
@@ -95,7 +97,7 @@ func dataSourceRead(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	return nil
 }
 
-func flattenBuckets(buckets []admin.DiskBackupSnapshotExportBucket) []map[string]any {
+func flattenBuckets(buckets []admin.DiskBackupSnapshotExportBucketResponse) []map[string]any {
```
Comment (maastha) [inline internal/service/cloudbackupsnapshotexportbucket/data_source_cloud_backup_snapshot_export_buckets.go]: change from the SDK, no impact

File: internal/service/cloudbackupsnapshotexportbucket/resource_cloud_backup_snapshot_export_bucket.go [hunk 4]
```diff
@@ -84,7 +86,7 @@ func resourceCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 
 	cloudProvider := d.Get("cloud_provider").(string)
 
-	request := &admin.DiskBackupSnapshotExportBucket{
+	request := &admin.DiskBackupSnapshotExportBucketRequest{
```
Comment (maastha) [inline internal/service/cloudbackupsnapshotexportbucket/resource_cloud_backup_snapshot_export_bucket.go]: change from the SDK, no impact

**Review by oarbusi — APPROVED** [review]
LGTM, and nice refactor! 
I just have a question, would it be possible to create a test with the non default value (false) of the new attribute?

File: internal/service/cloudbackupsnapshotexportbucket/data_source_cloud_backup_snapshot_export_buckets.go (hunk 3 above)
Comment (marcosuma) [reply internal/service/cloudbackupsnapshotexportbucket/data_source_cloud_backup_snapshot_export_buckets.go]: I think [this context](https://mongodb.slack.com/archives/CLB2CJ1JB/p1729539408447839) can be useful

**Review by marcosuma — APPROVED** [review]

**Review by erabil-mdb — APPROVED** [review]
Docs LGTM!

**Comment by maastha** [comment]
>I just have a question, would it be possible to create a test with the non default value (false) of the new attribute?

@oarbusi could you elaborate? There is an existing test that creates with the non-default value, see
https://github.com/mongodb/terraform-provider-mongodbatlas/blob/1acc3c7d843913c68ecdb10775e32801cbad6514/internal/service/organization/resource_organization_test.go#L102

File: internal/service/cloudbackupsnapshotexportbucket/data_source_cloud_backup_snapshot_export_buckets.go (hunk 3 above)
Comment (maastha) [reply internal/service/cloudbackupsnapshotexportbucket/data_source_cloud_backup_snapshot_export_buckets.go]: nice, thank you!

**Comment by oarbusi** [comment]
> > I just have a question, would it be possible to create a test with the non default value (false) of the new attribute?
> 
> @oarbusi could you elaborate? There is an existing test that creates with the non-default value, see
> 
> https://github.com/mongodb/terraform-provider-mongodbatlas/blob/1acc3c7d843913c68ecdb10775e32801cbad6514/internal/service/organization/resource_organization_test.go#L102

Totally missed that test when reviewing, thanks!

**Review by lantoli — APPROVED** [review]

**Pull request merged**